src/config.rs

```diff
@@ -0,0 +1,177 @@
+use config as config_external_crate;
+use config_external_crate::{Config, File, FileFormat};
+use crossterm::event::KeyCode;
+use directories::ProjectDirs;
+use serde::Deserialize;
+use std::collections::HashMap;
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, Deserialize)]
+pub enum Action {
+    Up,
+    Down,
+    Left,
+    Right,
+    PageUp,
+    PageDown,
+    Quit,
+    Reset,
+    NextTab,
+    PrevTab,
+}
+
+#[derive(Clone, Debug)]
+pub struct AppConfig {
+    pub keymap: Keymap,
+}
+
+#[derive(Debug, Deserialize)]
+struct RawConfig {
+    keybindings: HashMap<String, Action>,
+}
+
+#[derive(Debug, Clone)]
+pub struct Keymap {
+    bindings: HashMap<KeyCode, Action>,
+    reverse: HashMap<Action, KeyCode>,
+}
+
+impl Keymap {
+    pub fn get_action(&self, code: KeyCode) -> Option<Action> {
+        self.bindings.get(&code).copied()
+    }
+
+    pub fn get_keycode_string(&self, action: Action) -> String {
+        match self.reverse.get(&action) {
+            Some(KeyCode::Char(c)) => format!("[{}]", c),
+            Some(other) => format!("[{:?}]", other),
+            None => "[???]".to_string(),
+        }
+    }
+}
+
+fn default_keybindings() -> Keymap {
+    let mut bindings = HashMap::new();
+
+    bindings.insert(KeyCode::Char('k'), Action::Up);
+    bindings.insert(KeyCode::Char('j'), Action::Down);
+    bindings.insert(KeyCode::Char('h'), Action::Left);
+    bindings.insert(KeyCode::Char('l'), Action::Right);
```

Comment:
Could we also bind the arrow keys to these actions to provide backwards compatibility?